Assets/Scripts/UI/EMCli.cs

```diff
@@ -0,0 +1,184 @@
+        scrollPosition += new Vector2(0, 150);
+    }
+
+    public void EMCliGUI()
+    {
+        GUI.backgroundColor = Color.green;
+        windowRect = GUILayout.Window(208, windowRect, func1, "EM Command Line", options1);
+    }
+    
+    public static void WindowLayout(int windowID)
+    {
+        GUI.backgroundColor = Color.green;
+        GUILayout.BeginHorizontal();
+        if (GUILayout.Button(">", GUILayout.MaxWidth(25)))
+        {
+            try
+            {
+                EnterCommand(inputLine);
+            }
+            catch (Exception ex)
+            {
+                $" {ex.ToString()} ".SendError(true);
+            }
+            
+            inputLine = string.Empty;
+
+            return;
+        }
+        GUI.SetNextControlName("CLI");
+        inputLine = GUILayout.TextField(inputLine, new GUILayoutOption[0]);
+        if (GUILayout.Button("X".RepaintError(true), GUILayout.MaxWidth(25)))
+        {
+            inputLine = string.Empty;
+        }
+        GUILayout.EndHorizontal();
+
+        if ((Event.current.isKey))
+        {
+            if (Event.current.type == EventType.KeyUp)
+            {
+                //Event.current.keyCode.ToString().SendCli();
+                if ((Event.current.keyCode == KeyCode.Return) && GUI.GetNameOfFocusedControl().Equals("CLI"))
+                {
+                    if (!string.IsNullOrEmpty(inputLine))
+                    {
+                        try
+                        {
+                            EnterCommand(inputLine);
+                        }
+                        catch (Exception ex)
+                        {
+                            $" {ex.ToString() }".SendError(true);
+                        }
+                        inputLine = string.Empty;
+                        GUI.FocusControl(string.Empty);
+                        return;
+                    }
+                }
+            }
+        }
+
+        scrollPosition = GUILayout.BeginScrollView(scrollPosition);
+        GUILayout.Label(layout);
+        GUILayout.EndScrollView();
+
+        //GUILayout.Button(ScrollPosition.ToString());
+
+        GUI.DragWindow();
+    }
+
+    public static void EnterCommand(string command)
+    {
+        if (command.StartsWith("/"))
+        {
+            if (command.StartsWith("/sp"))
+            {
+                AottgUi.TestSpawn();
+            }
+            else if (command.StartsWith("/cn"))
+            {
+                FengGameManagerMKII.instance.StartCoroutine(EMCli.ConnectAndJoinIE(false));
+                FengGameManagerMKII.showHackMenu = false;
+            }
+            else
+            {
+                $"[{command}] - there is no such command!".SendError(true);
+            }
+        }
+        else
+        {
+            $"[{command}] is not a command!".SendError(true);
+        }
+    }
+
+    public static IEnumerator ConnectAndJoinIE(bool spawnAfterLoad)
```

Comment:
This method is nice, but it does throw ~200 error messages "JoinOrCreateRoom failed. Client is not on Master Server or not yet ready to call operations. Wait for callback: OnJoinedLobby or OnConnectedToMaster.
UnityEngine.Debug:LogError(Object)". I assume that this is just temporary?